test/unit/BondingManager.js

```diff
@@ -1921,20 +1942,25 @@ contract("BondingManager", accounts => {
             })
 
             it("fires a TranscoderSlashed event, but transcoder is not penalized because it does not have a bonded amount", async () => {
-                bondingManager.TranscoderSlashed({transcoder: transcoder}).on("data", e => {
-                    assert.equal(e.returnValues.finder, constants.NULL_ADDRESS, "should fire TranscoderSlashed event with null finder")
-                    assert.equal(e.returnValues.penalty, 0, "should fire TranscoderSlashed event with slashed amount penalty of 0")
-                    assert.equal(e.returnValues.finderReward, 0, "should fire TranscoderSlashed event with finder reward of 0")
-                })
-
-                await fixture.verifier.execute(
+                const txRes = await fixture.verifier.execute(
                     bondingManager.address,
                     functionEncodedABI(
                         "slashTranscoder(address,address,uint256,uint256)",
                         ["address", "uint256", "uint256", "uint256"],
                         [transcoder, constants.NULL_ADDRESS, PERC_DIVISOR / 2, 0]
                     )
                 )
+
+                // 'expectEvent' is used to parse deep events that don't occur in the base contract being called
+                // These events are not included as decoded event logs in a transaction receipt's 'logs' field
+                // Rather they are still encoded as a hex string in the 'rawLogs' field
+                expectEvent(
```

Comment:
I thought the helper was pretty neat because it simply decodes the raw event logs according to the abi rather than calling `bondigManager.getPastEvents()` but I can restructure it to overwrite the decoded event logs on the transaction receipt , then return the new receipt.

That way we can do

```
truffleAssert.eventEmitted(
    decodeRawLogs(BondingManager.abi, txRes),
    "TranscoderSlashed",
    e => e.transcoder == transcoder && e.finder == constants.NULL_ADDRESS && e.penalty == 0 && e.finderReward == 0
)
```

I'm fine with either option though.

Reply:
> rather than calling bondigManager.getPastEvents()

Ah didn't know that `truffleAssert.createTransactionResult()` uses `getPastEvents()` - thanks for the tip! It uses a filter that targets the specific block number of the provided transaction receipt though so its not as bad.

> it simply decodes the raw event logs according to the abi

I do like this approach! However, I think the helper will run into some trouble if the contract that returns `txRes` also emits an event as a part of the transaction. Take the `GenericMock` example and suppose `GenericMock` emitted an `Executed` event for `execute()`. Then, the receipt's `rawLogs` would include the `TranscoderSlashed` event AND the `Executed` event. Since `expectFail` takes in the ABI for `BondingManager` it will try to decode the `Executed` event and fail since that event is not a part of the provided ABI. This isn't an issue right now because `execute()` doesn't emit any events of its own, but `expectFail()` wouldn't be a generic drop-in unless additional changes are made to ensure that all events in `rawLogs` can be decoded regardless of what contract emitted them.

IMO just using `truffleAssert` for now feels simpler - WDYT?

Reply:
That's correct that this helper currently doesn't forsee decoding for multiple ABI's.

I'm okay with using `truffleAssert`.

Reply:
fixed in ad3cb40